cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -1287,7 +1406,28 @@ aggregate_reader_metadata::select_columns(
     for (auto& col : selected_columns) {
       auto const& top_level_col_schema_idx = find_schema_child(root, col.name);
       bool valid_column = build_column(&col, top_level_col_schema_idx, output_columns, false);
-      if (valid_column) output_column_schemas.push_back(top_level_col_schema_idx);
+      if (valid_column) {
+        output_column_schemas.push_back(top_level_col_schema_idx);
+
+        // Map the column's schema_idx across the rest of the data sources if required.
+        if (per_file_metadata.size() > 1 and schema_idx_maps.size()) {
+          std::for_each(thrust::make_counting_iterator(1ul),
```

Comment:
Let’s be consistent with the types. I think size_t is more clear than unsigned long.
```suggestion
          std::for_each(thrust::make_counting_iterator(static_cast<size_t>(1)),
```